packages/ra-core/src/core/Resource.tsx

```diff
@@ -92,32 +92,14 @@ const ResourceRoutes = (props: ResourceProps) => {
                         />
                     )}
                     {show && (
-                        <Route
-                            path={`${basePath}/:id/show`}
-                            render={routeProps => (
-                                <WithPermissions
-                                    component={show}
-                                    {...routeProps}
-                                />
-                            )}
-                        />
+                        <Route path={`${basePath}/:id/show`}>
+                            <WithPermissions component={show} />
+                        </Route>
                     )}
                     {edit && (
-                        <Route
-                            path={`${basePath}/:id`}
-                            render={routeProps => (
-                                <WithPermissions
-                                    component={edit}
-                                    basePath={basePath}
-                                    id={decodeURIComponent(
```

Comment:
I think we lost something here: what happens if the identifier contains URI encodable characters? We should add a unit test for that.